kafka-ui-api/src/main/java/com/provectus/kafka/ui/cluster/util/JmxClusterUtil.java

```diff
@@ -24,26 +26,46 @@
 
     private static final String JMX_URL = "service:jmx:rmi:///jndi/rmi://";
     private static final String JMX_SERVICE_TYPE = "jmxrmi";
+    private static final String KAFKA_SERVER_PARAM = "kafka.server";
 
-    public static final String BYTES_IN_PER_SEC = "BytesInPerSec";
-    public static final String BYTES_OUT_PER_SEC = "BytesOutPerSec";
-    private static final String BYTES_IN_PER_SEC_MBEAN_OBJECT_NAME = "kafka.server:type=BrokerTopicMetrics,name=" + BYTES_IN_PER_SEC;
-    private static final String BYTES_OUT_PER_SEC_MBEAN_OBJECT_NAME = "kafka.server:type=BrokerTopicMetrics,name=" + BYTES_OUT_PER_SEC;
-
-    private static final List<String> attrNames = Arrays.asList("OneMinuteRate", "FiveMinuteRate", "FifteenMinuteRate");
+    public List<JmxMetric> getJmxMetrics(int jmxPort, String jmxHost) {
+        String jmxUrl = JMX_URL + jmxHost + ":" + jmxPort + "/" + JMX_SERVICE_TYPE;
+        List<JmxMetric> result = new ArrayList<>();
+        JMXConnector srv = null;
+        try {
+            srv = pool.borrowObject(jmxUrl);
+            MBeanServerConnection msc = srv.getMBeanServerConnection();
+            var jmxMetrics = msc.queryNames(null, null).stream().filter(q -> q.getCanonicalName().startsWith(KAFKA_SERVER_PARAM)).collect(Collectors.toList());
+            jmxMetrics.forEach(j -> {
+                JmxMetric metric = new JmxMetric();
+                metric.setCanonicalName(j.getCanonicalName());
+                metric.setValue(getJmxMetric(jmxPort, jmxHost, j.getCanonicalName()));
+                result.add(metric);
+            });
+            pool.returnObject(jmxUrl, srv);
+        } catch (IOException ioe) {
+            log.error("Cannot get jmxMetricsNames, {}", jmxUrl, ioe);
+            closeConnectionExceptionally(jmxUrl, srv);
+        } catch (Exception e) {
+            log.error("Cannot get JmxConnection from pool, {}", jmxUrl, e);
+            closeConnectionExceptionally(jmxUrl, srv);
+        }
+        return result;
+    }
 
-    public Map<String, Number> getJmxTrafficMetrics(int jmxPort, String jmxHost, String metricName) {
+    private Map<String, Object> getJmxMetric(int jmxPort, String jmxHost, String canonicalName) {
         String jmxUrl = JMX_URL + jmxHost + ":" + jmxPort + "/" + JMX_SERVICE_TYPE;
-        Map<String, Number> result = new HashMap<>();
+
+        Map<String, Object> resultAttr = new HashMap<>();
```

Comment:
I think it's better to collect metrics once, instead of doing it cluster-wide and then per broker